### PR TITLE
Sdk 2257 net retrieve qr code

### PR DESCRIPTION
**Pull request by mehmet-yoti**

### CODE REVIEW

**Comment by irotech** [comment]
Shouldn't this PR be created against `SDK-2252-net-create-qr-code` instead of `development`?

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 1]
```diff
@@ -58,16 +58,86 @@ internal static async Task<GetSessionResult> GetSession(HttpClient httpClient, U
...
+
+                var responseObject = await response.Content.ReadAsStringAsync();
+                var deserialized = await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<CreateQrResult>(responseObject));
+
+                return deserialized;
+
+            }
+        }
+
+        internal static async Task<GetSessionResult> GetQrCode(HttpClient httpClient, Uri apiUrl, string sdkId, AsymmetricCipherKeyPair keyPair, string qrCodeId)
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:128]: GetSessionResult is not the expected response for the GET endpoint. Please check the openAPI docs as this endpoint has 2 different SUCCESS responses (depending on the header Yoti-Share-Mobile-Token)
for web
```
{
  "id": "qr.v2.TCR1yNPVTECiGxyH95ix4g",
  "expiry": "2023-20-07T02:00:12Z",
  "policy": "CgoKBnNlbGZpZSgBEhkKFWNvdWxkLWJlLWFueXRoaW5nLXRiaCgBGAEgAQ==",
  "extensions": [],
  "session": {
    "id": "ss.v2.ChZyZUJzOXFTNFRlV1lSa3ZhMkJBN3F3EgdsZDUuZ2Jy",
    "status": "QR_CODE_SCANNED",
    "expiry": "2023-20-07T02:00:12Z"
  },
  "redirectUri": "/redirect"
}
```
for mobile
```
{
  "id": "qr.v2.TCR1yNPVTECiGxyH95ix4g",
  "expiry": "2023-20-07T02:00:12Z",
  "policy": "CgoKBnNlbGZpZSgBEhkKFWNvdWxkLWJlLWFueXRoaW5nLXRiaCgBGAEgAQ==",
  "extensions": [],
  "session": {
    "id": "ss.v2.ChZyZUJzOXFTNFRlV1lSa3ZhMkJBN3F3EgdsZDUuZ2Jy",
    "expiry": "2023-20-07T02:00:12Z"
  },
  "shareToken": "_jmBVDmqQuKuZVG9K9DRZw",
  "domain": "app.com",
  "app": {
    "id": "b9adcd92-bf90-44cf-9204-ca1a6d4c1ecd",
    "name": "AppName",
    "trusted": true
  },
  "org": {
    "name": "OrgName",
    "status": "VERIFIED"
  }
}
```

Comment (saurabh-yoti) [reply src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:128]: Please correct the return type for `GetQrCode` task.

**Review by saurabh-yoti — CHANGES_REQUESTED** [review]

**Comment by irotech** [comment]
can you please rebase this branch and solve conflicts?

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 2]
```diff
@@ -125,5 +124,37 @@ internal static async Task<CreateQrResult> CreateQrCode(HttpClient httpClient, U
...
 
+        internal static async Task<GetQrCodeResult> GetQrCode(HttpClient httpClient, Uri apiUrl, string sdkId, AsymmetricCipherKeyPair keyPair, string qrCodeId)
+        {
+            Validation.NotNull(httpClient, nameof(httpClient));
+            Validation.NotNull(apiUrl, nameof(apiUrl));
+            Validation.NotNull(sdkId, nameof(sdkId));
+            Validation.NotNull(keyPair, nameof(keyPair));
+            Validation.NotNull(qrCodeId, nameof(qrCodeId));
+
+            Request shareSessionlRequest = new RequestBuilder()
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:135]: `fetchQrCode` or `fetchShareQrCode` or `fetchSessionQrCode` or something related to the QR?

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs [hunk 3]
```diff
@@ -125,5 +124,37 @@ internal static async Task<CreateQrResult> CreateQrCode(HttpClient httpClient, U
...
+                .WithEndpoint(string.Format($"/v2/qr-codes/{0}", qrCodeId))
+                .WithQueryParam("appId", sdkId)
+                .WithHttpMethod(HttpMethod.Get)
+                .Build();
+
+            using (HttpResponseMessage response = await shareSessionlRequest.Execute(httpClient).ConfigureAwait(false))
+            {
+                if (!response.IsSuccessStatusCode)
+                {
+                    Response.CreateYotiExceptionFromStatusCode<DynamicShareException>(response);
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs:148]: not quite sure this is a `DynamicShare`

File: src/Yoti.Auth/DigitalIdentity/GetQrResult.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
+﻿using Newtonsoft.Json;
+
+
+namespace Yoti.Auth.DigitalIdentity
+{
+    public class GetSessionResult
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/GetQrResult.cs:6]: this file name is `GetQrResult.cs` but the class name is `GetSessionResult`. Can u explain?

File: src/Yoti.Auth/DigitalIdentity/GetSessionResult.cs [hunk 5]
```diff
@@ -1,48 +1,32 @@
-﻿using Newtonsoft.Json;
-
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+using Yoti.Auth.DigitalIdentity.Extensions;
 
 namespace Yoti.Auth.DigitalIdentity
 {
-    public class GetSessionResult
+    public class GetQrCodeResult
```
Comment (irotech) [inline src/Yoti.Auth/DigitalIdentity/GetSessionResult.cs:7]: in here the file name is `GetSessionResult.cs` and the class name `GetQrCodeResult`

**Review by irotech — CHANGES_REQUESTED** [review]

File: src/Yoti.Auth/DigitalIdentity/GetSessionResult.cs (hunk 5 above)
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/GetSessionResult.cs]: File names were mixed and corrected now

File: src/Yoti.Auth/DigitalIdentity/GetQrResult.cs (hunk 4 above)
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/GetQrResult.cs]: File names were mixed and corrected now

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs (hunk 3 above)
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs]: you are right. Updated exception type

File: src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs (hunk 2 above)
Comment (mehmet-yoti) [reply src/Yoti.Auth/DigitalIdentity/DigitalIdentityService.cs]: corrected request variable names

**Review by irotech — APPROVED** [review]
LGTM from the API POV

**Review by saurabh-yoti — APPROVED** [review]

**Pull request merged**